channel/src/main/java/com/synopsys/integration/alert/channel/jira/common/distribution/custom/JiraCustomFieldResolver.java

```diff
@@ -92,6 +92,28 @@ protected Object convertValueToRequestObject(CustomFieldDefinitionModel fieldDef
         }
     }
 
+    private JsonArray createJsonArray(String innerFieldValue, String fieldArrayItems) {
+        JsonArray jsonArray = new JsonArray();
+        switch (fieldArrayItems) {
+            case CUSTOM_FIELD_TYPE_STRING_VALUE:
+                jsonArray.add(innerFieldValue);
+                break;
+            case CUSTOM_FIELD_TYPE_COMPONENT_VALUE:
+                for (String componentName : StringUtils.split(innerFieldValue)) {
+                    jsonArray.add(createJsonObject("name", componentName));
+                }
+                break;
+            case CUSTOM_FIELD_TYPE_OPTION_VALUE:
+                for (String optionValue : StringUtils.split(innerFieldValue)) {
+                    jsonArray.add(createJsonObject("value", optionValue));
+                }
+                break;
+            default:
+                throw new AlertRuntimeException(String.format("Unsupported item: '%s' for array field type", fieldArrayItems));
+        }
+        return jsonArray;
+    }
```

Comment:
I heard the plan was get rid of switch-cases, but as long as they are in our code, let's try to avoid `break;` statements as much as possible. It's too easy to make mistakes with them. Instead we can break this into two methods: 

1. `JsonArray createJsonArray(String innerFieldValue, String fieldArrayItems)`
2. `List<Object> createJsonArrayElements(String innerFieldValue, String fieldArrayItems)`

The second method should only have `return` and `throw` statements.